lucille-core/src/test/java/com/kmwllc/lucille/core/DocumentTest.java

```diff
@@ -1654,6 +1655,83 @@ public void testRemoveChildren() {
+        .set("b", mapper.createObjectNode()
+            .set("c", mapper.createObjectNode()
+                .set("d", mapper.createObjectNode()
+                    .put("e", 234))));
+    document.setNestedJson(("a.b.c.d"), mapper.createObjectNode().put("e", 234));
+    assertEquals(jsonNode, document.getJson("a"));
+    assertEquals(234, document.getNestedJson("a.b.c.d.e").asInt());
+
+    // test nested field that contains a list in a top level field
+    JsonNode jsonList = mapper.createArrayNode()
+        .add(mapper.createObjectNode()
+                .put("name", "thing1")
+                .put("age", 50))
+        .add(mapper.createObjectNode()
+                .put("name", "thing2")
+                .put("age", 25));
+    document.setNestedJson("list.0", mapper.createObjectNode().put("name", "temp").put("age", 87));
+    document.setNestedJson("list.1", mapper.createObjectNode().put("name", "thing2").put("age", 25));
+    document.setNestedJson("list.0", mapper.createObjectNode().put("name", "thing1").put("age", 50));
+    assertEquals(jsonList, document.getJson("list"));
+    assertEquals("thing2", document.getNestedJson("list.1.name").asText());
+
+    // test nested field with a list that is more nested
+    JsonNode moreNestedJsonList = mapper.createObjectNode().set("list", mapper.createArrayNode()
+        .add(mapper.createObjectNode()
+            .put("name", "thing1")
+            .put("age", 50))
+        .add(mapper.createObjectNode()
+            .put("name", "thing2")
+            .put("age", 25)
+            .set("phone", mapper.createObjectNode()
+                .put("number", "123-4567")
+                .put("areaCode", "303"))));
+    document.setNestedJson("doc.list.0", mapper.createObjectNode().put("name", "thing1").put("age", 50));
+    document.setNestedJson("doc.list.1", mapper.createObjectNode().put("name", "thing2").put("age", 25));
+    document.setNestedJson("doc.list.1.phone", mapper.createObjectNode().put("number", "123-4567").put("areaCode", "303"));
+    assertEquals(moreNestedJsonList, document.getJson("doc"));
+    assertEquals("thing2", document.getNestedJson("doc.list.1.name").asText());
+
+    // test non nested field
+    document.setNestedJson("simple", mapper.createObjectNode().put("field", "simpleValue"));
+    assertEquals("simpleValue", document.getNestedJson("simple.field").asText());
+
+    // test non-existing field
+    assertNull(document.getNestedJson("a.does.not.exist"));
+  }
+
+  @Test
+  public void testNestedArrayUpdate() {
+    Document doc = Document.create("id1");
+    doc.setNestedJson("a.b.0", IntNode.valueOf(50));
+    doc.setNestedJson("a.b.0", IntNode.valueOf(60));
+    assertEquals(1, doc.getNestedJson("a.b").size());
+    assertEquals(60, doc.getNestedJson("a.b.0").intValue());
+    assertNull(doc.getNestedJson("a.b.1"));
+  }
+
+  @Test
+  public void testNestedArrayUpdateConfusingArrayIndexScenario() throws IllegalArgumentException {
```

Comment:
minor point re the name of this test -- I'm not sure this scenario is "confusing" any more... it was confusing back when two updates at index 1 would result in an array with two values...

maybe testNestedArrayUpdateIndexOutOfBounds() .. or roll these tests into the above testNestedArrayUpdate()